src/middleware/server.rs

```diff
@@ -39,6 +39,7 @@ where
         signal_hook::flag::register(signal_hook::consts::SIGINT, Arc::clone(&stop))?;
         signal_hook::flag::register(signal_hook::consts::SIGTERM, Arc::clone(&stop))?;
 
+        let mut metric_data = Vec::new();
```

Comment:
instead of this, i tried this:

```
struct Metric<'a> {
    raw: &'a [u8]
}
```

and Metric would be constructed directly from `raw` with no copying